Source/Core/UICommon/ResourcePack/Manager.cpp

```diff
@@ -31,35 +31,46 @@ IniFile GetPackConfig()
 bool Init()
 {
   packs.clear();
-  auto pack_list = Common::DoFileSearch({File::GetUserPath(D_RESOURCEPACK_IDX)}, {".zip"});
-
-  bool error = false;
+  std::vector<std::string> pack_list =
+      Common::DoFileSearch({File::GetUserPath(D_RESOURCEPACK_IDX)}, {".zip"});
 
   IniFile file = GetPackConfig();
 
   auto* order = file.GetOrCreateSection("Order");
 
-  std::sort(pack_list.begin(), pack_list.end(), [order](std::string& a, std::string& b) {
-    std::string order_a = a, order_b = b;
+  struct OrderHelper
+  {
+    size_t pack_list_index;
+    std::string manifest_id;
+  };
 
-    order->Get(ResourcePack(a).GetManifest()->GetID(), &order_a);
-    order->Get(ResourcePack(b).GetManifest()->GetID(), &order_b);
+  std::vector<OrderHelper> pack_list_order;
+  pack_list_order.reserve(pack_list.size());
+  for (size_t i = 0; i < pack_list.size(); ++i)
+  {
+    ResourcePack pack(pack_list[i]);
+    pack_list_order.emplace_back(
+        OrderHelper{i, pack.IsValid() ? pack.GetManifest()->GetID() : pack_list[i]});
+  }
 
-    return order_a < order_b;
-  });
+  std::sort(pack_list_order.begin(), pack_list_order.end(),
+            [order](const OrderHelper& a, const OrderHelper& b) {
+              return a.manifest_id < b.manifest_id;
+            });
 
-  for (size_t i = 0; i < pack_list.size(); i++)
+  bool error = false;
+  for (size_t i = 0; i < pack_list_order.size(); ++i)
   {
-    const auto& path = pack_list[i];
+    const auto& path = pack_list[pack_list_order[i].pack_list_index];
 
-    if (!Add(path))
+    ResourcePack* pack = Add(path);
+    if (!pack)
```

Comment:
```suggestion
    ResourcePack* const pack = Add(path);
    if (pack != nullptr)
```

Reply:
`pack == nullptr` you mean. I don't think `!ptr` is that unusual but sure, whatever.

Reply:
Oops, yeah.

My impression was that explicitly comparing against nullptr was preferred, but not a big deal either way.